src/transformers/generation_logits_process.py

```diff
@@ -702,3 +702,31 @@ class LogitNormalization(LogitsProcessor, LogitsWarper):
     def __call__(self, input_ids: torch.Tensor, scores: torch.Tensor) -> torch.Tensor:
         scores = scores.log_softmax(dim=-1)
         return scores
+
+
+class SuppressBlank(LogitsProcessor):
+    r""" """
+
+    def __init__(self, blank_token_id, eos_token_id, sample_begin: int = 1):
+        self.blank_token_id = blank_token_id
+        self.eos_token_id = eos_token_id
+        self.sample_begin = sample_begin
+
+    def __call__(self, input_ids, scores):
+        tokens = input_ids
+        logits = scores
+        if tokens.shape[1] == self.sample_begin:
+            logits[:, self.blank_token_id + [self.eos_token_id]] = -np.inf
+        return logits
+
+
+class SuppressTokens(LogitsProcessor):
```

Comment:
Note for whisper they should be called automatically -> this means you'll also have to add a and a `suppress_tokens: Optional[List[int]]`  function argument to `generate` as well as to the config (even if we deprecate it soon @gante)